pkg/resource/plugin/host.go

```diff
@@ -198,6 +209,72 @@ func (host *defaultHost) Analyzer(name tokens.QName) (Analyzer, error) {
 	return plugin.(Analyzer), nil
 }
 
+func (host *defaultHost) PolicyAnalyzer(name tokens.QName, path string) (Analyzer, error) {
+	plugin, err := host.loadPlugin(func() (interface{}, error) {
+		// First see if we already loaded this plugin.
+		if plug, has := host.analyzerPlugins[name]; has {
+			contract.Assert(plug != nil)
+			return plug.Plugin, nil
+		}
+
+		// If not, try to load and bind to a plugin.
+		plug, err := NewPolicyAnalyzer(host, host.ctx, name, path)
+		if err == nil && plug != nil {
+			info, infoerr := plug.GetPluginInfo()
+			if infoerr != nil {
+				return nil, infoerr
+			}
+
+			// Memoize the result.
+			host.plugins = append(host.plugins, info)
+			host.analyzerPlugins[name] = &analyzerPlugin{Plugin: plug, Info: info}
+		}
+
+		return plug, err
+	})
+	if plugin == nil || err != nil {
+		return nil, err
+	}
+	return plugin.(Analyzer), nil
+}
+
+func (host *defaultHost) ListAnalyzers() []Analyzer {
+	analyzers := []Analyzer{}
```

Comment:
Seems like we might want to cache this list

Reply:
Hm, I suppose it isn't static over the lifetime of the host.

Reply:
Hmm, what do you have in mind specifically?

Reply:
Once an update begins, no new analyzer plugins will be registered, right? If that is the case, then the plan should be able to capture this list up-front in order to avoid recomputing it each time a resource is registered. It's not super important, but it stuck out to me.

Reply:
I like it the way it is now because it's really close to the way `ListPlugins` works -- it returns the list of analyzers the host knows about. The difference is that it projects that list from `[]Analyzer` -> `[]Plugin` (_i.e._, with the loop). If this is important to you we can change it but that's my view.

Reply:
It isn't terribly important, no.